cranelift/codegen/src/verifier/mod.rs

```diff
@@ -1918,14 +1918,14 @@ impl<'a> Verifier<'a> {
                 arg,
                 ..
             }
-            | ir::InstructionData::InsertLane {
+            | ir::InstructionData::TernaryImm8 {
                 opcode: ir::instructions::Opcode::Insertlane,
-                lane,
+                imm: lane,
                 args: [arg, _],
                 ..
             } => {
                 // We must be specific about the opcodes above because other instructions are using
-                // the ExtractLane/InsertLane formats.
+                // the same formats.
```

Comment:
I'm surprised to see we don't use `0x6b`/`0x69`-encoded `IMUL`, where `TernaryImm8` would be a better-fitting format in fact (it's a three-operand encoding in the form `imul rrr, r/m, imm8`, or `imm32` for the 32/64-bit forms). Do you know what other instructions are referenced here? A read through the legalizer seems to suggest this isn't actually true.

Reply:
answer: `x86_pshufd` where the immediate is not a lane selector, it's a mask of how to shuffle (where there is no encoding for a type-invalid mask that we would want to check in this arm)

and this comment will become more true in the future with additional instructions taking mask immediates.